xcat3/common/rest.py

```diff
@@ -0,0 +1,65 @@
+from oslo_log import log as logging
+from xcat3.common import exception
+
+import requests
+import json
+
+LOG = logging.getLogger(__name__)
+
+class RestSession :
+
+    headers = {'Content-Type': 'application/json'}
+
+    def __init__(self) :
+        self.session = requests.Session()
+
+    def _request_log(self, method, url, data) :
+        log_string = ['curl -k']
+        log_string.append(' -X %s' % method)
+        log_string.append(' -H %s' % RestSession.headers)
+        log_string.append(" '%s'" % url)
+
+        if data :
+            log_string.append('-d %s' % data)
+
+        LOG.info("REQ: %s" % "".join(log_string))
+
+    def _rspcheck(self, rsp) :
+        if rsp.text and rsp.status_code != 400 :
+            try :
+                body = json.loads(rsp.text)
+            except ValueError :
+                body = None
+        else :
+            body = None
+
+        LOG.info("RESP: [%(status)s]\nRESP BODY: %(text)s\n",
+                {'status': rsp.status_code, 'text': json.dumps(body)})
+
+        if rsp.status_code >= 400 :
+            rsp_data = rsp.json()
+            raise exception.ExceptionFromRestRsp(exception=rsp_data['data']['description'])
```

Comment:
As this library is running in the daemon, it will be used to send different request to different servers. If error happens, we do not know what the url is , how can we know is it a problem of our client request or the problem of the server?

If you can get these message from response, it is not necessary to put the method and url in the exception message.